pkg/cosign/keys.go

```diff
@@ -105,6 +106,18 @@ func PemToECDSAKey(pemBytes []byte) (*ecdsa.PublicKey, error) {
 	return ecdsaPub, nil
 }
 
+func PemToRSAKey(pemBytes []byte) (*rsa.PublicKey, error) {
```

Comment:
What do you think of merging these functions so we can do a type assertion switch instead of trying them in order as error fallbacks?

Reply:
Looking at top of sigstore/sigstore project for helper apis. Just thinking about this some more, to support a full verify list, more hash funcs may need to be checked against if done "automatically" without cmdline arg hints. For example, cosign assumes sha256, but sha512 is another possibility.

Reply:
Updated the PR using sigstore/sigstore helpers.
Caveat is we (continue to) assume SHA256.